bulkExport/glueScripts/export-script.py

```diff
@@ -0,0 +1,124 @@
+    type = getResolvedOptions(sys.argv, ['type'])['type']
+groupId = None
+if ('--{}'.format('groupId') in sys.argv):
+    groupId = getResolvedOptions(sys.argv, ['groupId'])['groupId']
+
+job_id = args['jobId']
+export_type = args['exportType']
+transaction_time = args['transactionTime']
+since = args['since']
+outputFormat = args['outputFormat']
+glue_table_name = args['glueTableName']
+
+glue_database = args['glueDatabase']
+bucket_name = args['s3OutputBucket']
+
+# Read data from DDB
+original_data_source_dyn_frame = glueContext.create_dynamic_frame.from_catalog(database = glue_database, table_name = glue_table_name)
+
+print('Start filtering by transactionTime and Since')
+# Filter by transactionTime and Since
+datetime_since = datetime.strptime(since, "%Y-%m-%dT%H:%M:%S.%fZ")
+datetime_transaction_time = datetime.strptime(transaction_time, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+filtered_dates_dyn_frame = Filter.apply(frame = original_data_source_dyn_frame,
+                           f = lambda x:
+                           datetime.strptime(x["meta"]["lastUpdated"], "%Y-%m-%dT%H:%M:%S.%fZ") > datetime_since and
+                           datetime.strptime(x["meta"]["lastUpdated"], "%Y-%m-%dT%H:%M:%S.%fZ") <= datetime_transaction_time
+                          )
+
+print('Start filtering by documentStatus and resourceType')
+# Filter by resource listed in Type and with correct STATUS
+type_list = None if type == None else type.split(',')
+valid_document_state_to_be_read_from = ['AVAILABLE','LOCKED', 'PENDING_DELETE']
+filtered_dates_resource_dyn_frame = Filter.apply(frame = filtered_dates_dyn_frame,
+                                    f = lambda x:
+                                    x["documentStatus"] in valid_document_state_to_be_read_from if type_list is None
+                                    else x["documentStatus"] in valid_document_state_to_be_read_from and x["resourceType"] in type_list
+                          )
+total_records_to_export = filtered_dates_resource_dyn_frame.count()
+print(f'Total records to export {total_records_to_export}')
+if total_records_to_export > 0:
+    print('Dropping fields that are not needed')
+    # Drop fields that are not needed
+    data_source_cleaned_dyn_frame = DropFields.apply(frame = filtered_dates_resource_dyn_frame, paths = ['documentStatus', 'lockEndTs', 'vid'])
+
+    data_frame = data_source_cleaned_dyn_frame.toDF()
+    # If only a few records, we should combine the output to 1 file per resource
+    # For lots of record, combining all records into 1 file per resource causes memory out of limit error in Glue
+    # Each record is about 2kb, so 50,000 is about 100MB of data
+    if (total_records_to_export < 50000):
+        print('Combining data into fewer partitions')
+        data_frame = data_frame.coalesce(1)
+
+    # Create duplicated column so we can use it in partitionKey later
+    data_frame = data_frame.withColumn('resourceTypeDup', data_frame.resourceType)
```

Comment:
When using `partitionKey` in line 95 it removes the field from the record, therefore I'm creating a duplicate of `resourceType` so that I can use `partitionKey` later on to split records into corresponding `resourceType` files. 

[link](https://aws.amazon.com/blogs/big-data/work-with-partitioned-data-in-aws-glue/)
"When you execute this write, the type field is removed from the individual records and is encoded in the directory structure. "